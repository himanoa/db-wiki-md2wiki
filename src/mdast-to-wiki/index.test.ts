import { describe, it, expect } from 'vitest'
import { toWikiPlugin, toWikiText } from '.'
import remarkParse from 'remark-parse'
import {unified} from 'unified'
import { readFileSync } from 'fs'
import { join } from 'path'

describe("toWikiText", () => {
  it('should be plain text', () => {
    expect(toWikiText()({ type: 'text', value: 'fooo' })).toStrictEqual("fooo")
  })

  it('should be blockquote text', () => {
    expect(toWikiText()({ type: 'blockquote', children: [{
      type: 'paragraph',
      children: [
        {
          type: 'text',
          value: 'fooo'
        }
      ]
    }]})).toStrictEqual("<blockquote>fooo\n</blockquote>")
  })

  it('should be codeblock text', () => {
    expect(toWikiText()({ type: 'code', value: 'foo\nbar'})).toStrictEqual(` foo
 bar
`)
  })

  it('should be heading text', () => {
    expect(toWikiText()({ type: 'heading', depth: 1, children:[ {
      type: 'text',
      value: 'foo'
    } ]})).toStrictEqual(`= foo =\n`)

    expect(toWikiText()({ type: 'heading', depth: 2, children:[ {
      type: 'text',
      value: 'foo'
    } ]})).toStrictEqual(`== foo ==\n`)

    expect(toWikiText()({ type: 'heading', depth: 3, children:[ {
      type: 'text',
      value: 'foo'
    } ]})).toStrictEqual(`=== foo ===\n`)

    expect(toWikiText()({ type: 'heading', depth: 4, children:[ {
      type: 'text',
      value: 'foo'
    } ]})).toStrictEqual(`==== foo ====\n`)

    expect(toWikiText()({ type: 'heading', depth: 5, children:[ {
      type: 'text',
      value: 'foo'
    } ]})).toStrictEqual(`===== foo =====\n`)

    expect(toWikiText()({ type: 'heading', depth: 6, children:[ {
      type: 'text',
      value: 'foo'
    } ]})).toStrictEqual(`====== foo ======\n`)
  })

  it("should be ordered list", () => {
    expect(toWikiText()({
      type: 'list',
      ordered: true,
      children: [
        {
          type: 'listItem',
          children: [
            {
              type: 'paragraph',
              children: [{
                type: 'text',
                value: 'foo'
              }]
            }
          ]
        },
        {
          type: 'listItem',
          children: [
            {
              type: 'paragraph',
              children: [{
                type: 'text',
                value: 'bar'
              }]
            }
          ]
        }
      ]
    })).toMatchInlineSnapshot(`
      "# foo
      # bar
      "
    `)

    expect(toWikiText()({
      type: 'list',
      ordered: true,
      children: [
        {
          type: 'listItem',
          children: [
            {
              type: 'paragraph',
              children: [
                {
                  type: 'text',
                  value: 'foo'
                }
              ],
            },
            {
              type: 'list',
              ordered: true,
              children: [{
                type: 'listItem',
                children: [
                  {
                    type: 'paragraph',
                    children: [
                      { type: 'text', value: 'nested' }
                    ]
                  }
                ]
              }]
            }
          ]
        },
        {
          type: 'listItem',
          children: [
            {
              type: 'paragraph',
              children: [{
                type: 'text',
                value: 'bar'
              }]
            }
          ]
        }
      ]
    })).toMatchInlineSnapshot(`
      "# foo
      ## nested
      # bar
      "
    `)

    expect(toWikiText()({
      type: 'list',
      ordered: false,
      children: [
        {
          type: 'listItem',
          children: [
            {
              type: 'paragraph',
              children: [
                {
                  type: 'text',
                  value: 'foo'
                }
              ],
            },
            {
              type: 'list',
              ordered: false,
              children: [{
                type: 'listItem',
                children: [
                  {
                    type: 'paragraph',
                    children: [
                      { type: 'text', value: 'nested' }
                    ]
                  }
                ]
              }]
            }
          ]
        },
        {
          type: 'listItem',
          children: [
            {
              type: 'paragraph',
              children: [{
                type: 'text',
                value: 'bar'
              }]
            }
          ]
        }
      ]
    })).toMatchInlineSnapshot(`
      "* foo
      ** nested
      * bar
      "
    `)
  })

  it("should be strong", () => {
    expect(toWikiText()({
      type: 'strong',
      children: [{
        type: 'text',
        value: 'foo'
      }]
    })).toStrictEqual("'''foo'''")
  })

  it("should be italic", () => {
    expect(toWikiText()({
      type: 'emphasis',
      children: [{
        type: 'text',
        value: 'foo'
      }]
    })).toStrictEqual("''foo''")
  })

  it("integration test", async () => {
    const body = readFileSync(join(__dirname, 'example-patch-note.md'), 'utf8')
    const processed = String(await unified().use(remarkParse).use(toWikiPlugin).process(body))
    expect(processed).toMatchSnapshot()
  })
})
