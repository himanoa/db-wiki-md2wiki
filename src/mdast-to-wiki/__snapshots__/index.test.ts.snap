// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`toWikiText > integration test 1`] = `
"= アーリーアクセス パッチノート（2024年8月15日） =
冒険者の皆様へ、
<t:1723708800:F>から<t:1723716000:F>までサーバーメンテナンスが予定されていることをご了承ください。
メンテナンスの40分前の<t:1723706400:F>からマッチメイキングキューは閉鎖されます。進行中のマッチには影響がありません。
== 　新しいコンテンツ ==
エピックとレジェンドの間に、エクセプショナル品質の装備が追加されました。プレイヤーはダンジョンやクロートーの試練で入手し、より強力な装備を使用して敵と戦うことができます！エアルームも新しいエクセプショナル品質の装備を作ることができます！
新しいPVPプレイモード「クロートーの試練」は8月16日に正式開始されます。期間中に初めて「完璧」の称号を獲得すれば、オレンジ色の装備や限定のキルエフェクトを獲得できます！世界中からのプロたちと競い合い、装備と技術を披露しましょう。今回はクロートーの試練に関連するSteamの実績も追加されています！
「自然」テーマの星座イベントが開始され、自然の知恵に通じたドルイドが新しい代替スキル「快癒」を獲得し、チーム戦術に新たな選択肢をもたらしました。同時に新しい期間限定テーマイベントも開催されていますので、早く限定の外見報酬を入手しましょう！
商店には新しい呪骨シリーズの武器と防具の外見が追加されました。恐怖の騎士（Chosen of Terror）に変身し、敵に血と恐怖が交錯したデスメタルを奏でましょう！
== 　クラスバランス ==
###　ローグ
ローグを影から現れるキャラクターとして描くことを望んでいますが、現在の選出率と脱出率が高すぎます、特に単独プレイ時に。悪質な回避行為やチーミングがますます深刻になっており、ゴールドファーマーのローグは他の普通にプレイするプレイヤーのゲーム体験に深刻な影響を与えています。この更新では、消極的な回避行為をするローグがより大きなコストと代償を払い、他のクラスにローグを逮捕する機会を与えます。
ゲーム内で受けるすべてのダメージ効果により、隠れる効果が解除されます。魔法の浸食（サークル）も含まれます。
隠れている目標がより簡単に見つかるようになりました。15メートル以内で可視化になります。
器用さによる移動速度ボーナスを低下させました。
インタラクション速度上げのステータスは、テレポートを開くのに影響を与えなくなりました。インタラクション速度上げのステータスの上限効果は、最大50％の加速効果を提供するようになりました。
Eスキル「クイックスニーク」のクールダウンが5秒増加しました。「隠匿」のクールダウンが10秒増加しました。
クイックスニークのクールダウンメカニズムが、隠れている状態を解除した後にCDに入るように変更されました。
隠れているスキルのCD中に再びパッシブにより隠れると、CDがより長くなる問題が修正されました。
この変更がもたらすデータの変化を継続的に監視します。同時に、積極的に戦う単独プレイヤーが予想どおりのゲーム体験に戻るようにするために、以下のマッチメイキング戦略の調整を行います。現在、同じクラスが多すぎる場合、より長い待ち時間になる可能性があります。
=== 　ファイター ===
ファイターの対策が徐々に改善されるにつれて、ファイターたちの陣地を突破する能力は環境に対して徐々に不利になっています。この更新では、ファイタープレイヤーに少しの助けを提供し、競争力のある前線の突破者になることを期待しています。
パッシブ「激昂」の効果が変更され、Eスキルを2回使用できるようになります。
旋風斬りを回転させる過程で、ファイターは徐々に移動速度が上昇します。
=== 　ドルイド ===
この更新では、ドルイドは新しい代替スキルを獲得し、治療能力を持つ2番目のクラスとなりました。プリーストとは異なり、ドルイドはHOT治療形式に重点を置き、自身の機動性を発揮しながらチームに強力な持続力を提供できます。チーム編成により多くの選択肢をもたらすことを期待しています。
新しいEスキル「快癒」が追加されました：味方に「快癒」を施し、30秒間生命力を回復します。
Eスキル「自然の力」が「自然の種」パッシブを発動した後、2回のスキル発動間隔が5秒から1秒に調整されました。
ドルイドのトレントが敵に攻撃すると、クリティカルヒットが発生するようになりました。
=== 　パイロマンサー ===
Qスキル「ファイアボール」の第二段階は、隠れている状態でも観察されるようになりました。
=== 　クライオマンサー ===
Eスキル「アイスバリア」の持続時間を8秒から5秒、回復効果を50％低減させることに調整されました。
* 暗闇ダメージを与える片手剣を使用する際、片手剣オーブの近接DPSがやや高すぎます。さらに、片手剣オーブの近接チャージ攻撃は、盾を素早く破壊し、減速し、バーストダメージを与える能力を兼ね備えており、近接キャラクターがアイスメイジ以降の安定した利益を得られなくなるほどのコストを負担しています。このため、以下の調整を行いました：
* 片手剣オーブの通常攻撃のダメージ倍率を0.6、0.6、1から0.4、0.4、0.6に調整
* 片手剣オーブのエンチャントによる追加氷元素ダメージ倍率を0.3から0.4に調整
* 上記の2つの変更で、より大きな割合のダメージをエンチャントに移し、エレメンタルソードと物理ソードのダメージ差を縮小し、暗闇片手剣オーブの近接持続出力能力を低下させました
* 片手剣オーブのチャージ攻撃のダメージ倍率を0.75から0.5に低下し、発射される投射物のダメージ倍率を0.5から0.3に低下
=== 　プリーストについて ===
過去1週間で、プリーストの上位試合での選出率と脱出率が着実に上昇しており、プリーストの強化が遅れています。命中HP回復の削減後のデータ状況を引き続き監視し、正しい判断を下すための確認を行います。
== 　 アイテムバランス ==
* ミミックポーション：他のプレイヤーに簡単に偽の宝箱として見破られることに悩んでいますか？今ではさまざまな種類の宝箱に変身し、ランダムにスイッチと光を生成し、対戦相手にビックリさせましょう！
* 復活の石のメカニズム調整：各祭壇は1回だけ復活の石をリフレッシュでき、品質はエピックに変更
* 雷のフラスコと焔のフラスコは投擲目標点で有効範囲指示特効を追加し、攻撃範囲をより効果的に判断できるようになりました
* 焔のフラスコ
-ダメージを受けると同時に重傷効果が付与され、この効果によりすべての回復を50%の低下させます。
** 火焔の上に立つと0.15秒ごとに5ダメージを与えます
** 火焔から離れると、焼けると重傷効果が8秒持続し、0.5秒ごとに5ダメージを与えます
== 　他のバランス調整 ==
* より多くのプレーヤーデータ統計を拡張し、更新後にバッグ画面で表示できるようにしました
* パーセント減速メカニズムを乗算演算に変更し、速度を25%未満に低下させなくなりました。これにより加速と絶対値の減速に影響はありません
* 期間限定のシャドウポータル確率増加イベントが終了し、シャドウポータルは通常の出現確率に戻りました
* 悪用を防ぐため、オークションハウスの購入機能にレベル制限を追加しました。現在、10レベル以上のキャラクターを所有するか、【オークションハウスリスティング特権】を持っているアカウントが使用できます。（この変更はトレード機能の使用には影響しません）
== 　その他 ==
* レジェンド武器「過ち」のモデルに新しい特殊効果を追加しました
* 神格の剣に氷のエンチャントの視覚効果を追加し、魔法クラスがこの武器をより効果的に振るうことができるようになりました
* レジェンドおよびエピック装備の同質化取引入札上限を引き上げました
* クライオマンサーの「ブリザード」のスキルエフェクトを最適化し、展開後の氷の嵐の境界をより明確にしました
* トレードのフィルタリングロジックを最適化し、更新後、オークションハウスではメインステータスとサブステータスを同時にフィルタリングできるようになりました
* 美術シーンの効果を最適化し、プレイヤーが遠くにいるときのシーンの状態が正確でない問題を解決しました
* モンスターの美術表現を最適化し、一部のモンスターモデルとハイライトの誤りを修正しました
* 一部のスキルアイコンを最適化し、スタイルと色調をより統一しました
* マントの布の効果を最適化しました
* アイスメイジのスキルと特性に関連するローカライズの説明を最適化しました
* セキュリティコンポーネントの打撃能力を最適化し、重大な不正行為を監視し、処罰の力を強化しました
* メイス盾の防御攻撃判定を最適化しました
* マップ中の一部の不合理なリスポーン地点の位置を最適化しました
* 一部の戦闘中のサウンドエフェクトを最適化しました
-画面設定には、DLSSフレーム生成オプションが追加されました。NVIDIA RTX 40シリーズのグラフィックカードを使用しているプレイヤーは、最新のフレーム生成技術を使用するために設定で有効にできます。
##　 バグ修正
* キー設定の衝突を引き起こす問題を修正し、キー設定のオプションを最適化しました
* 盾の防御攻撃判定の遅延を引き起こす問題を修正しました
* ドルイドのジャガーから人間形態への変身、アバターの視認不可、透明化の問題を修正しました
* 特定の場所でシーンから抜け出すことができる問題を修正しました
* レイフォースタッフのチャージスキル指示が階段や床を貫通する問題を修正しました
* プレイヤーが特定の場所でスタックし、移動できない問題を修正しました
* ローグが隠れる状態から複数の敵を倒してパッシブの隠れる状態を再起動し、Eスキルの同時CD再開の問題を修正しました
* ポーションや包帯を使用する際にアイテム欄がロックされない問題を修正しました
* 祝祷をアクティブ化した後、地下二階待機室に入ると蓄積回数が誤って表示される問題を修正しました
* クライオマンサーのQスキルである「ブリザード」が一部のシーンで移動が異常な問題を修正しました
* 復讐の王がモンスターを召喚する際にダメージを受けた後、剣を持った位置が異常な問題を修正しました
* 遠くから見るとポータルが楕円形になる問題を修正しました
* 遠くから見るとポータルが表示されない問題を修正しました
* リコネクションの進捗バーが立ち往生する可能性のある問題を修正しました
* クロスボウが照準を切り替える際に困難なゾンビ毒霧攻撃に襲われ、毒霧のエフェクトが異常な問題を修正しました
* テキストが重なる可能性のある問題を修正しました
* チームメイトのエネルギーバーが復活後に異常に表示される問題を修正しました
* すべての装備を脱いでトレーニングエリアに入ると、バッグのプレビューが異常になる問題を修正しました
* PVP復活後にインタラクションができない問題を修正しました
"
`;
